sklearn/neural_network/tests/test_mlp.py

```diff
@@ -1046,3 +1047,40 @@ def test_mlp_sample_weight_with_early_stopping():
 
     m2 = MLPRegressor(**params).fit(X, y, sample_weight=None)
     assert_allclose(m1.predict(X), m2.predict(X))
+
+
+def test_mlp_vs_poisson_glm_equivalent(global_random_seed):
+    """Test MLP with Poisson loss and no hidden layer equals GLM."""
+    n = 100
+    rng = np.random.default_rng(global_random_seed)
+    X = np.linspace(0, 1, n)
+    y = rng.poisson(np.exp(X + 1))
+    X = X.reshape(n, -1)
+    glm = PoissonRegressor(alpha=0, tol=1e-7).fit(X, y)
+    # Unfortunately, we can't set a zero hidden_layer_size, so we use a trick by using
+    # just one hidden layer node with an identity activation. Coefficients will
+    # therefore be different, but predictions are the same.
+    mlp = MLPRegressor(
+        loss="poisson",
+        hidden_layer_sizes=(1,),
+        activation="identity",
+        alpha=0,
+        solver="lbfgs",
+        tol=1e-7,
+        random_state=np.random.RandomState(global_random_seed + 1),
+    ).fit(X, y)
+
+    assert_allclose(mlp.predict(X), glm.predict(X), rtol=1e-5)
+
+    # The same does not work with the squared error because the output activation is
+    # the idendity instead of the exponential.
+    mlp = MLPRegressor(
+        loss="squared_error",
+        hidden_layer_sizes=(1,),
+        activation="identity",
+        alpha=0,
+        solver="lbfgs",
+        tol=1e-7,
+        random_state=np.random.RandomState(global_random_seed + 1),
+    ).fit(X, y)
+    assert not np.allclose(mlp.predict(X), glm.predict(X), rtol=1e-4)
```

Comment:
Do we really need to check for the squared_error not matching here? Asking because this test seems to be related mainly to the poisson loss.

Reply:
If squared error based prediction were the same, we might be in the interpolation regime.
Also, it shows more clearly that the poisson loss is of added value.